### PR TITLE
additional tests for the safe base64 decoder

### PR DESCRIPTION
**Pull request by lemire**
cc @constellation @pauldreik

### CODE REVIEW

**Comment by pauldreik** [comment]
this looks like a good candidate to fuzz, but I need to finish what I started first....

File: fuzz/roundtrip.cpp [hunk 1]
```diff
@@ -496,6 +496,36 @@ extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
         }
       }
     }
+
```
Comment (pauldreik) [inline fuzz/roundtrip.cpp:499]: I think the roundtrip fuzzer covers too many cases already, it is a bit hard to grasp when it is too long. Can it be in a separate file? I can have a look at it later once I finish what I have in flight.

Comment (lemire) [reply fuzz/roundtrip.cpp:499]: Sure. We can break it down. This can be done as a follow-up PR. :-)

**Comment by lemire** [comment]
@pauldreik merged but we can improve the QOL by breaking things down. 👍

**Pull request merged**